src/GraphQL/Types/Composite/AutoRegisteringInterfaceGraphType.cs

```diff
@@ -0,0 +1,154 @@
+        {
+            foreach (var memberInfo in GetRegisteredMembers())
+            {
+                bool include = true;
+                foreach (var attr in memberInfo.GetGraphQLAttributes())
+                {
+                    include = attr.ShouldInclude(memberInfo, false);
+                    if (!include)
+                        break;
+                }
+                if (!include)
+                    continue;
+                var fieldType = CreateField(memberInfo);
+                if (fieldType != null)
+                    yield return fieldType;
+            }
+        }
+
+        /// <inheritdoc cref="AutoRegisteringObjectGraphType{TSourceType}.CreateField(MemberInfo)"/>
+        protected virtual FieldType? CreateField(MemberInfo memberInfo)
+        {
+            var typeInformation = GetTypeInformation(memberInfo);
+            var graphType = typeInformation.ConstructGraphType();
+            var fieldType = AutoRegisteringHelper.CreateField(memberInfo, graphType, false);
+            BuildFieldType(fieldType, memberInfo);
+            // apply field attributes after resolver has been set
+            AutoRegisteringHelper.ApplyFieldAttributes(memberInfo, fieldType, false);
+            return fieldType;
+        }
+
+        /// <inheritdoc cref="AutoRegisteringObjectGraphType{TSourceType}.BuildFieldType(FieldType, MemberInfo)"/>
+        protected void BuildFieldType(FieldType fieldType, MemberInfo memberInfo)
+        {
+            Func<Type, Func<FieldType, ParameterInfo, ArgumentInformation>> getTypedArgumentInfoMethod =
+                parameterType =>
+                {
+                    var getArgumentInfoMethodInfo = _getArgumentInformationInternalMethodInfo.MakeGenericMethod(parameterType);
+                    return (Func<FieldType, ParameterInfo, ArgumentInformation>)getArgumentInfoMethodInfo.CreateDelegate(typeof(Func<FieldType, ParameterInfo, ArgumentInformation>), this);
+                };
+
+            AutoRegisteringHelper.BuildFieldTypeForOutput(
+                memberInfo,
+                fieldType,
+                BuildMemberInstanceExpression,
+                getTypedArgumentInfoMethod,
+                ApplyArgumentAttributes);
+        }
+
+        /// <inheritdoc cref="AutoRegisteringObjectGraphType{TSourceType}.BuildMemberInstanceExpression(MemberInfo)"/>
+        protected virtual LambdaExpression BuildMemberInstanceExpression(MemberInfo memberInfo)
+            => _sourceExpression;
+
+        private static readonly Expression<Func<IResolveFieldContext, TSourceType>> _sourceExpression
+            = context => (TSourceType)(context.Source ?? ThrowSourceNullException());
+
+        private static object ThrowSourceNullException()
+        {
+            throw new NullReferenceException("IResolveFieldContext.Source is null; please use static methods when using an AutoRegisteringInterfaceGraphType as a root graph type or provide a root value.");
```

Comment:
Throwing `NullReferenceException` is not a recommended practice, NRE is a runtime exception in fact and should not be thrown by any user code.

Reply:
Changed to InvalidOperationException